layouts/_default/single.html

```diff
@@ -9,51 +9,32 @@
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "item") }}
-      {{ partial "fragments/item.html" (dict "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "table") }}
-      {{ partial "fragments/table.html" (dict "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "embed") }}
-      {{ partial "fragments/embed.html" (dict "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "member") }}
-      {{ partial "fragments/member.html" (dict "resources" $.Resources "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "logos") }}
-      {{ partial "fragments/logos.html" (dict "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "contact") }}
-      {{ partial "fragments/contact.html" (dict "Site" .Site "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (in .Params.fragment "footer") }}
-      {{ partial "fragments/footer.html" (dict "menus" $.Site.Menus "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "copyright") }}
-      {{ partial "fragments/copyright.html" (dict "menus" $.Site.Menus "Site" .Site "self" . ) }}
-    {{ end }}
-
-    {{ if and (not .Params.disabled) (eq .Params.fragment "buttons") }}
-      {{ partial "fragments/buttons.html" (dict "self" . ) }}
+  {{/*
+    This code may not be understandable immediately. This is trying to find all
+    the resources (read fragments) for each page and all the global fragments.
+    When combining global and local fragments, local ones have more priority.
+    So if there is only one footer fragment and that is global, that footer
+    fragment is shown. The same is true if there are no global footer fragments
+    and one local one. But if there is one global footer fragment and one local
+    footer fragment, the local one is rendered.
+    */}}
+  {{ $global := .Site.GetPage "page" "global" }}
+  {{ $global_fragments := $global.Resources.ByType "page" }}
+  {{ $local_fragments := .Resources.ByType "page" }}
+
+  {{ $.Scratch.Delete "fragments" }}
+  {{ range $local_fragments }}
+    {{ $.Scratch.Add "fragments" (slice .) }}
+  {{ end }}
+  {{ range $global_fragments }}
+    {{ if not (where ($.Scratch.Get "fragments") ".Name" .Name) }}
+      {{ $.Scratch.Add "fragments" (slice .) }}
     {{ end }}
+  {{ end }}
 
-    {{ if and (not .Params.disabled) (in .Params.fragment "custom") }}
-      {{ partial (print "fragments/" .Params.fragment_name ".html") (dict "self" .) }}
+  {{ range sort ($.Scratch.Get "fragments") "Params.weight" }}
+    {{ if and (not .Params.disabled) (isset .Params "fragment") (not (eq .Params.fragment "hero")) }}
+      {{ partial (print "fragments/" .Params.fragment ".html") (dict "menus" $.Site.Menus "Site" $.Site "resources" $.Resources "self" .) }}
```

Comment:
I would have assumed we used $.Scratch in the fragments so that not providing "$" or "tmp_full" would break hero, table and item.

Reply:
Considering this we might actually just move to "$" and "." in this PR. Thoughts?

Reply:
Let's continue this one in the other discussion https://github.com/okkur/syna/pull/108#discussion_r187721597